internal/daemon/controller/listeners.go

```diff
@@ -276,13 +276,13 @@ func (c *Controller) configureForCluster(ln *base.ServerListener) (func(), error
 		}
 		go func() {
 			err := splitListener.Start()
-			if err != nil {
+			if !errors.Is(err, net.ErrClosed) {
 				event.WriteError(c.baseContext, op, err, event.WithInfoMsg("splitListener.Start() error"))
 			}
 		}()
 		go func() {
 			err := ln.GrpcServer.Serve(metric.InstrumentClusterTrackingListener(multiplexingAuthedListener, grpcListenerPurpose))
-			if err != nil {
+			if !errors.Is(err, net.ErrClosed) {
```

Comment:
Same here, should this also check for `nil`? The docs for `grpc.(Server).Serve` state:
> Serve will return a non-nil error unless Stop or GracefulStop is called.

So it seems it will return nil if stopped.

Reply:
It does seem like that should be the case, but existing behavior seems to not agree:
```
{"id":"i9CIGvYZ3D","source":"https://hashicorp.com/boundary/arch/worker","specversion":"1.0","type":"error","data":{"error":"use of closed network connection","error_fields":{},"id":"e_8Jo7RdIvuk","version":"v0.1","op":"controller.configureForCluster","info":{"msg":"multiplexingAuthedListener error"}},"datacontentype":"application/cloudevents","time":"2023-09-20T13:40:26.300154787-04:00"}
```
I didn't chase this down too much other than it's the grpc server set on the cluster listener and we seem to be calling GracefulStop on it.